src/Exception/Exception.php

```diff
@@ -56,78 +58,90 @@ class Exception extends \Exception
+	public function __construct(
+		array|string $args = [], // @deprecated
+
+		string|null $key = null,
+		array|null $data = null,
+		array|null $details = null,
+		string|null $fallback = null,
+		int|null $httpCode = null,
+		string|null $message = null,
+		Throwable|null $previous = null,
+		bool $translate = true
+	) {
+		$key      ??= $args['key'] ?? null;
+		$fallback ??= $args['fallback'] ?? null;
+		$previous ??= $args['previous'] ?? null;
+
+		$this->data =
+			$data ??
+			$args['data'] ??
+			static::$defaultData;
+
+		$this->httpCode =
+			$httpCode ??
+			$args['httpCode'] ??
+			static::$defaultHttpCode;
+
+		$this->details =
+			$details ??
+			$args['details'] ??
+			static::$defaultDetails;
 
-		// define the Exception key
-		$key = $args['key'] ?? static::$defaultKey;
+		// set the Exception code to the key
+		$this->code = $key ?? static::$defaultKey;
 
-		if (Str::startsWith($key, self::$prefix . '.') === false) {
-			$key = self::$prefix . '.' . $key;
+		if (Str::startsWith($this->code, self::$prefix . '.') === false) {
+			$this->code = self::$prefix . '.' . $this->code;
 		}
 
 		if (is_string($args) === true) {
+			$message ??= $args;
+		}
+
+		if ($message !== null) {
 			$this->isTranslated = false;
-			parent::__construct($args);
-		} else {
-			// define whether message can/should be translated
-			$translate =
-				($args['translate'] ?? true) === true &&
-				class_exists(App::class) === true;
-
-			// fallback waterfall for message string
-			$message = null;
-
-			if ($translate === true) {
-				// 1. translation for provided key in current language
-				// 2. translation for provided key in default language
-				if (isset($args['key']) === true) {
-					$message = I18n::translate(self::$prefix . '.' . $args['key']);
-					$this->isTranslated = true;
-				}
-			}
-
-			// 3. provided fallback message
-			if ($message === null) {
-				$message = $args['fallback'] ?? null;
-				$this->isTranslated = false;
-			}
-
-			if ($translate === true) {
-				// 4. translation for default key in current language
-				// 5. translation for default key in default language
-				if ($message === null) {
-					$message = I18n::translate(self::$prefix . '.' . static::$defaultKey);
-					$this->isTranslated = true;
-				}
-			}
-
-			// 6. default fallback message
-			if ($message === null) {
-				$message = static::$defaultFallback;
-				$this->isTranslated = false;
-			}
-
-			// format message with passed data
-			$message = Str::template($message, $this->data, ['fallback' => '-']);
-
-			// handover to Exception parent class constructor
-			parent::__construct($message, 0, $args['previous'] ?? null);
+			return parent::__construct($message);
 		}
 
-		// set the Exception code to the key
-		$this->code = $key;
+		// define whether message can/should be translated
+		$translate = $args['translate'] ?? $translate;
+
+		// fallback waterfall for message string
+		$message = null;
```

Comment:
I think `$message` is always `null` for this line.

Reply:
This is to ensure that `$message` exists in any case as it then will be filled inside some if conditions. Otherwise it would not exist for cases that match no if statement.

Reply:
I think it always exists since we have `$message` named argument and it's `null` since we have `$message !== null` check in above line.

Reply:
👍 now understood it, yes